app/samples/src/main/java/com/sygic/samples/search/viewmodels/SearchFromBrowseMapWitchPinsActivityViewModel.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.samples.search.viewmodels
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import com.sygic.maps.module.common.provider.ModuleConnectionProvider
+import com.sygic.maps.module.search.SearchFragment
+import com.sygic.maps.uikit.viewmodels.common.extensions.loadDetails
+import com.sygic.maps.uikit.views.common.extensions.asSingleEvent
+import com.sygic.maps.uikit.views.common.livedata.SingleLiveEvent
+import com.sygic.maps.uikit.views.common.utils.logInfo
+import com.sygic.sdk.map.MapRectangle
+import com.sygic.sdk.map.`object`.MapMarker
+import com.sygic.sdk.position.GeoBoundingBox
+import com.sygic.sdk.position.GeoCoordinates
+import com.sygic.sdk.search.*
+import com.sygic.sdk.search.detail.DetailPoiCategory
+import com.sygic.sdk.search.detail.DetailPoiCategoryGroup
+
+private const val MARGIN = 80
+
+class SearchFromBrowseMapWitchPinsActivityViewModel : ViewModel(), ModuleConnectionProvider {
+
+    val addMapMarkerObservable: LiveData<MapMarker> = SingleLiveEvent()
+    val removeAllMapMarkersObservable: LiveData<Any> = SingleLiveEvent()
+    val setCameraPositionObservable: LiveData<GeoCoordinates> = SingleLiveEvent()
+    val setCameraRectangleObservable: LiveData<MapRectangle> = SingleLiveEvent()
+    val setCameraZoomLevelObservable: LiveData<Float> = SingleLiveEvent()
+
+    private val callback: ((searchResultList: List<SearchResult>) -> Unit) = { searchResultList ->
+        removeAllMapMarkersObservable.asSingleEvent().call()
+
+        if (searchResultList.isNotEmpty()) {
+            if (searchResultList.isOnlyCategory()) {
```

Comment:
`isOnly` znamena ze je iba 1 result?

Reply:
uz vidim

Reply:
nenol by lepsi nazov `isCategoryResult`, to `isOnlyCategory` sice chapem, ale je to take krkolomne

Reply:
oki, přejmenuji ;)